src/extensions/default/CodeFolding/Prefs.js

```diff
@@ -85,8 +93,9 @@ define(function (require, exports, module) {
       * @param {Object} folds the fold ranges to save for the current document
       */
     function setFolds(path, folds) {
-        store[path] = simplify(folds);
-        prefs.set(foldsKey, store);
+        var allFolds = PreferencesManager.getViewState(foldsKey);
```

Comment:
@peterflynn  can you make the **|| {}** change in your PR?
Also issues#10869 added for tracking the bucketing by project.